skfp/model_selection/splitters/pubchem_split.py

```diff
@@ -0,0 +1,470 @@
+        during a request.
+
+    n_jobs : int, default=5
+        The number of jobs to run in parallel for fetching data from PubChem. Note that
+        since those are requests relying on I/O, threads are used instead of processes.
+        At most 5 requests per second are made to avoid throttling, which limits
+        parallelism even if high ``n_jobs`` is set.
+
+    verbose : int, default=0
+        Controls the verbosity when fetching data from PubChem.
+
+    Returns
+    -------
+    subsets : tuple[list, list, ...]
+        Tuple with train-valid-test subsets of provided arrays. First three are lists of
+        SMILES strings or RDKit ``Mol`` objects, depending on the input type. If ``return_indices``
+        is True, lists of indices are returned instead of actual data.
+
+    References
+    ----------
+    .. [1] `Kim S, Thiessen PA, Cheng T, Yu B, Bolton EE.
+        "An update on PUG-REST: RESTful interface for programmatic access to PubChem."
+        Nucleic Acids Res. 2018 Jul 2;46(W1):W563-W570.
+        <https://doi.org/10.1093/nar/gky294>`_
+    """
+    years = _get_pubchem_years(data, n_jobs, n_tries, verbose)
+
+    data_df = pd.DataFrame({"year": years})
+
+    if not_found_behavior != "remove":
+        not_found_mols = list(data_df[data_df["year"].isna()].index)
+
+    data_df = data_df[~data_df["year"].isna()]
+
+    data_df = data_df.sort_values("year")
+
+    train_size, valid_size, test_size = validate_train_valid_test_split_sizes(
+        train_size, valid_size, test_size, len(data)
+    )
+
+    train_idxs: list[int] = data_df.index[:train_size].tolist()
+    valid_idxs: list[int] = data_df.index[train_size : train_size + valid_size].tolist()
+    test_idxs: list[int] = data_df.index[train_size + valid_size :].tolist()
+
+    if not_found_behavior == "train":
+        train_idxs.extend(not_found_mols)
+    elif not_found_behavior == "test":
+        test_idxs.extend(not_found_mols)
+
+    if return_indices:
+        train_subset = train_idxs
+        valid_subset = valid_idxs
+        test_subset = test_idxs
+    else:
+        train_subset = get_data_from_indices(data, train_idxs)
+        valid_subset = get_data_from_indices(data, valid_idxs)
+        test_subset = get_data_from_indices(data, test_idxs)
+
+    ensure_nonempty_subset(train_subset, "train")
+    ensure_nonempty_subset(valid_subset, "validation")
+    ensure_nonempty_subset(test_subset, "test")
+
+    if additional_data:
+        additional_data_split = split_additional_data(
+            list(additional_data), train_idxs, valid_idxs, test_idxs
+        )
+        return train_subset, valid_subset, test_subset, *additional_data_split
+    else:
+        return train_subset, valid_subset, test_subset
+
+
+@validate_params({"n_jobs": [int, (1, 5)]}, prefer_skip_nested_validation=True)
+def _get_pubchem_years(
+    data: Sequence[Union[str, Mol]], n_jobs: int, n_tries: int, verbose: int
```

Comment:
`n_retries` everywhere, I think this should be a number of retries, this is more common